l1-verifier/contracts/OPL1Verifier.sol

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.17;
+
+import { IEVMVerifier } from "@ensdomains/evm-verifier/contracts/IEVMVerifier.sol";
+import { RLPReader } from "@eth-optimism/contracts-bedrock/src/libraries/rlp/RLPReader.sol";
+import { StateProof, EVMProofHelper } from "@ensdomains/evm-verifier/contracts/EVMProofHelper.sol";
+import { L1Block } from "@eth-optimism/contracts-bedrock/src/L2/L1Block.sol";
+
+struct L1WitnessData {
+    uint256 blockNo;
+    bytes blockHeader;
+}
+
+contract OPL1Verifier is IEVMVerifier {
+    error BlockHeaderHashMismatch(uint256 current, uint256 number, bytes32 expected, bytes32 actual);
+
+    L1Block public l1Block;
```

Comment:
unfortunately L1Block contract doesn't let you access previous block hashes, so there may be issues if the L1 gateway doesn't return fast enough, and the witness data it provides isn't the latest L1 block anymore

Reply:
https://gov.optimism.io/t/final-extend-the-l1block-contract-to-store-historical-block-hash-data/6103 should help once it gets merged!